code/modules/paperwork/stamps.dm

```diff
@@ -12,6 +12,68 @@
 	item_color = "cargo"
 	pressure_resistance = 2
 	attack_verb = list("stamped")
+	/// used for on human stamp marks, each mark will be colored this before adding to image in human stamp_marks
+	var/stamp_color = "#6ca151"
+	/// this is defined uniquely for each subtype, used in examine.dm for when humans are stamped.
+	var/stamp_description = "..."
+
+/obj/item/stamp/afterattack(atom/target, mob/user, proximity, params)
+	. = ..()
+	if(!ishuman(target) || !ishuman(user) || !proximity)
+		return
+
+	var/mob/living/carbon/human/H = target
+
+	if(length(H.stamp_marks.overlays) == 50)	//no more than 50 marks on a person, otherwise infinite laggy stamp marks
+		to_chat(user, "<span class='notice'>Bluespace mechanics prevent any more marks from appearing on [target.name].</span>")
+		return
+
+	var/attackedSide				//the stamp mark will appear on this side, NORTH = forwards, SOUTH = back, EAST = right, WEST = left
+	var/list/paramslist = params2list(params)
+	var/xOffset = text2num(paramslist["icon-x"]) - 16
```

Comment:
How does this react to when you click the icon in the alt+click panel in the statpane? Those screw with `icon-` params and it could result in stamp marks way off the sprite.

Reply:
![dreamseeker_peeKgjs1tY](https://user-images.githubusercontent.com/16618648/102521042-09c3e700-405a-11eb-938c-40530cf0910b.png)
I believe because they are icons that are initially offset then overlayed on an image and then finally that is overlayed on a mutable appearance in update_icons.dm it doesn't respond poorly? Only issue I see with them in that menu is that the coloration of them is lost, which although an issue Its one thats shared with all bloody sprites for humans.

Reply:
I mean clicking the mob in that image with the stamp in hand, the `icon-x` and `icon-y` values get really screwed up.

Reply:
At worst they wont show up on the character and they will have their descriptions updated, my code cuts out any marks that won't appear on a base person with their clothes, hair, and tail. But from my testing it works very well.